File: 0007-reverse-integer/0007-reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:
        sign = 1 if x > 0 else -1
        num = sign*x
        reverse = 0
        
        while num != 0:
            reverse = reverse * 10 + num % 10
            num //= 10
        
        reverse *= sign
        
        if reverse > (2**31) - 1 or reverse < -2**31:
            return 0
        
        return reverse

File: 0007-reverse-integer/0007-reverse-integer.java
class Solution {
    public int reverse(int x) {
        int sign =x<0?-1:1;
        int num=sign*x;
        long reverse=0;

        
        while(num!=0){
            reverse=reverse*10+num%10;
            num/=10;
            if (reverse > Integer.MAX_VALUE || reverse < Integer.MIN_VALUE) {
                return 0;
            }
        }

        return sign*(int)reverse;
        
    }
}